packages/conversation/src/OpenAi.ts

```diff
@@ -38,6 +41,12 @@ export class OpenAi {
     const response = await OpenAi.executeRequest(messageParamsWithHistory, logLevel, functions, model);
     const responseMessage = response.choices[0].message;
     if (responseMessage.function_call) {
+      if (this.functionCallCount >= maxFunctionCalls) {
+        logger.error(`Max function calls (${maxFunctionCalls}) reached. Stopping execution.`);
+        return `Max function calls (${maxFunctionCalls}) reached.`;
```

Comment:
as is, this would return this string as n3xa's response message. instead we should throw an error